superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigModal.test.tsx

```diff
@@ -17,198 +17,289 @@
+  expect(await screen.findByText('birth_names')).toBeInTheDocument();
+  userEvent.type(screen.getByRole('combobox'), `Column A${specialChars.enter}`);
+  userEvent.click(getCheckbox(DEFAULT_VALUE_REGEX));
+  await waitFor(() => {
     expect(
-      screen.getByText('Multiple select').parentElement!,
-    ).not.toHaveAttribute(
-      'class',
-      'ant-checkbox-wrapper ant-checkbox-wrapper-checked',
-    );
-
-    // choose default value
-    userEvent.click(await screen.findByText('3 options'));
-    userEvent.click(screen.getByTitle('Abigail'));
-
-    // fill scoping
-    userEvent.click(screen.getByText('Apply to specific panels'));
-    userEvent.click(screen.getByText('CHART_ID'));
-
-    // saving
-    userEvent.click(screen.getByText('Save'));
-    await waitFor(() =>
-      expect(onSave.mock.calls[0][0][0]).toEqual(
-        expect.objectContaining({
-          cascadeParentIds: [],
-          controlValues: {
-            defaultToFirstItem: false,
-            enableEmptyFilter: false,
-            inverseSelection: false,
-            multiSelect: false,
-            sortAscending: true,
-          },
-          defaultValue: ['Abigail'],
-          filterType: 'filter_select',
-          isInstant: false,
-          name: 'Select Filter 1',
-          scope: { excluded: [], rootPath: [] },
-          targets: [{ column: { name: 'name' }, datasetId: 11 }],
-        }),
-      ),
-    );
+      screen.queryByText(FILL_REQUIRED_FIELDS_REGEX),
+    ).not.toBeInTheDocument();
   });
+  expect(
+    await screen.findByText(DEFAULT_VALUE_REQUIRED_REGEX),
+  ).toBeInTheDocument();
+});
+
+test('validates the hierarchical value', async () => {
+  defaultRender();
+  userEvent.click(screen.getByText(ADVANCED_REGEX));
+  userEvent.click(getCheckbox(HIERARCHICAL_REGEX));
+  expect(await screen.findByText(PARENT_REQUIRED_REGEX)).toBeInTheDocument();
+});
+
+test('validates the pre-filter value', async () => {
+  defaultRender();
+  userEvent.click(screen.getByText(ADVANCED_REGEX));
+  userEvent.click(getCheckbox(PRE_FILTER_REGEX));
+  expect(
+    await screen.findByText(PRE_FILTER_REQUIRED_REGEX),
+  ).toBeInTheDocument();
 });
+
+/*
+  TODO
+    adds a new value filter type with all fields filled
+    adds a new numerical range filter type with all fields filled
+    adds a new time range filter type with all fields filled
+    adds a new time column filter type with all fields filled
+    adds a new time grain filter type with all fields filled
+    collapsible controls opens by default when it is checked
+    advanced section opens by default when it has an option checked
+    deletes a filter
+    disables the default value when default to first item is checked
+    changes the default value options when the column changes
+    switches to configuration tab when validation fails
+    displays cancel message when there are pending operations
+    do not displays cancel message when there are no pending operations
```

Comment:
```suggestion
    does not display cancel message when there are no pending operations
```
obviously just a nit... really appreciate this TODO comment!